taipy/core/common/_self_setter_set.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021-2024 Avaiga Private Limited
```

Comment:
💭 The purpose of this class is a bit unclear to me. Maybe some more explanations about what problem it is solving would help us to review it.

Reply:
ah you're right! the idea is simply to have a custom set class that will trigger auto saving the entity the changes to the repo everytime something is done to the set, like adding, removing, etc.

Reply:
I am not sure I approve that design...
"Autosaving capability", if we want it, should be the responsibility of the entity itself, not the type of collection used. Collections should not implement any logic of that kind IMHO.
Furthermore, Taipy code usually make use of decorators for such a thing... 
Why introduce a new concept here?

Reply:
because I want to utilize the functions available to a Python `set`. Without this class, you will need to load and make changes to the set, then overwrite the `blocked_jobs` (for example) with an entire new set. It's definitely doable but it's just weird to look at IMHO. Also error prone as we need to make sure we overwrite the `blocked_jobs` with the new set or else it won't be saved. Let me know if that makes sense to you